### PR TITLE
Settings - Display erroring bank notice on Settings > Payments > Deposits

### LINKED ISSUE #8331: Display notice on Settings > Payments > Deposits to inform the merchant when their bank account has an error.
### Description
Display a notice in Settings > Payments > Deposits to inform the merchant when their bank account has an error.
- Based on https://github.com/Automattic/woocommerce-payments/issues/6294#issuecomment-1977424883
- Similar to the notice displayed on Payments > Overview > Deposits #8321 
- Dependent on woocommerce-payments-server#4862

### Acceptance criteria
When Stripe sets a bank account to `errored` after a failed deposit, the merchant is informed via a notice on  Settings > Payments > Deposits

### Designs
TBD

### Testing instructions
- Test with woocommerce-payments-server#4862
- Visit the Stripe Express dashboard for your test WooPayments account (Payments → Overview → Account Details → Edit Details) and update the payout bank account to a [failing test account](https://href.li/?https://docs.stripe.com/connect/testing#payouts), e.g. 110000000/000111111116
- Use the [Stripe Shell](https://href.li/?https://docs.stripe.com/connect/manual-payouts?shell=true&api=true&resource=accounts&action=retrieve) to create a manual deposit.
stripe payouts create --stripe-account="acct_123" --amount="123" --currency="USD" 
- The bank account would be set to `errored`.

### Dev notes
https://github.com/Automattic/woocommerce-payments/pull/8321
https://github.com/Automattic/woocommerce-payments-server/pull/4862

### Additional context
TBD: how to test for multiple currency --> bank account setups.

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #8331 

#### Changes proposed in this Pull Request

The PR adds a notice to `Settings > Payments > Deposits` to inform the merchant whenever their bank account has an error. Check screenshot:

image

#### Testing instructions
* Visit the Stripe Express dashboard for your test WooPayments account (Payments → Overview → Account Details → Edit Details) and update the payout bank account to a [failing test account](https://href.li/?https://docs.stripe.com/connect/testing#payouts), e.g. 110000000/000111111116
* Use the [Stripe Shell](https://href.li/?https://docs.stripe.com/connect/manual-payouts?shell=true&api=true&resource=accounts&action=retrieve) to create a manual deposit.
stripe payouts create --stripe-account="acct_123" --amount="123" --currency="USD"
* The bank account would be set to errored.
* Check within `Settings > Payments > Deposits`, you will see an error notice similar to the one in the screenshot
* Edit the account on Stripe back to a normal test bank account. The error will disappear.

#### Notes
* Similar notice has also been added within `Payments > Overview > Deposits` via #8321 , and Deposits list screen via #8347
* Related server PR - https://github.com/Automattic/woocommerce-payments-server/pull/4862

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply) - Checked on Firefox and Chrome in mobile view via browser tools

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9482` or branch name `fix/8331-bank-error-notice-settings-deposit` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8331-bank-error-notice-settings-deposit) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 4e8a2ba5c499ec46d8602c708097223dc1e27069
- Build time: 2024-09-26 09:15:52 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/deposits/index.js [hunk 1]
```diff
@@ -208,6 +210,17 @@ const Deposits = () => {
...
+	const { account, overview } = useSelectedCurrencyOverview();
+	const selectedCurrency =
+		overview?.currency || wcpaySettings.accountDefaultCurrency;
+
+	const hasErroredExternalAccount =
+		account?.default_external_accounts?.some(
+			( externalAccount ) =>
+				externalAccount.currency === selectedCurrency &&
+				externalAccount.status === 'errored'
+		) ?? false;
```
Comment (Jinksi) [inline client/settings/deposits/index.js]: 💭 I found that when I used a multi-deposit-currency store, this notice did not display, and there is no way to see to change which currency is selected on the settings page.

What do you think about making this display if _any_ deposit currency's default deposit account is in an `errored` state?

```suggestion
	const { overviews } = useAllDepositsOverviews();

	const hasErroredExternalAccount =
		overviews.account?.default_external_accounts?.some(
			( externalAccount ) => externalAccount.status === 'errored'
		) ?? false;
```

**Review by Jinksi — APPROVED** [review]
Nice one @nagpai 🌟

I've tested this with a multi-deposit-currency store with a single `errored` deposit destination.

I've left a suggestion to show merchants this if _any_ deposit currency's destination is `errored`, rather than just the "selected" currency.

Other than that, this is a great addition for merchants on the Payments settings screen.

File: client/settings/deposits/index.js (hunk 1 above)
Comment (nagpai) [reply client/settings/deposits/index.js]: >What do you think about making this display if any deposit currency's default deposit account is in an errored state?

Thanks for testing this with a multi-currency account! what you shared here makes sense!

Comment (nagpai) [reply client/settings/deposits/index.js]: I have incorporated the feedback here - [e6afbda](https://github.com/Automattic/woocommerce-payments/pull/9482/commits/e6afbda8c99e5ad509af4332e9759731c8a3e07b)

Also updated the tests to include multi-currency default accounts scenario - [4e8a2ba](https://github.com/Automattic/woocommerce-payments/pull/9482/commits/4e8a2ba5c499ec46d8602c708097223dc1e27069)

Thanks once again, for the valuable review, Eric 🙌🏼

Comment (nagpai) [reply client/settings/deposits/index.js]: >I've tested this with a multi-deposit-currency store with a single errored deposit destination.

I did a round of manual test this time with a similar account, and it seems to work as expected.

**Pull request merged**